### ISSUE TITLE
Interfaces are not preceeded with a capital I - Hungarian notation doesn't comply with Java convetions

### ISSUE THREAD

**Issue by ghost** [MinecraftForge/MinecraftForge#6423]
Always bothered me since the first time I've seen it: Who ever came up with preceed interfaces with a captial I obvious didn't read the Java convetions.
Guess I don't have to explain the typical Java naming schemes as it's used all over the SE API - but for those who don't know:
interfaces are named with the base word: List
abstract base implementations get the prefix "Abstract": AbstractList
concrete implementations (often subclases of the abstract base implementation) either get names based on their internal implementation or get the suffix "Impl": ArrayList, LinkedList, "ListImpl"
TLDR: the whole MC modding community needs a complete refactor to get the naming right to fit typical Java naming conventions ... it just hurts the eyes of serious Java developers.
The inventors of Java didn't came up with conventions just cause they had free time to spent - but they planed some unified schemed wich should be used by everyone using the language to increase code readability.

**Comment by LexManos** [comment]
No, we have already had this discussion.
Class naming conventions are a personal opinion, you're free to use whatever system you want in your own mod.

**Comment by ghost** [comment]
As said: Java conventions weren't made up cause someone was bored but to make it easier to work in a group as anyone uses the same coding style. But just ignoring it and close down any suggestion to "do it the Java way" is also a way of looking at it ...

**Comment by LexManos** [comment]
We are not ignoring it, it was discussed, and voted on by the community. They elected to keep the I prefix.
Conventions are in place so that people working in a group can understand that is going on.
As long as those using the system can understand things, and within a single project you are consistent. It is fine.
It has nothing to do with EVERYONE using the same coding style.
Want to name all your classes after Disney princesses? Go for it. You are free to do whatever you want in your own project. 
The Minecraft community has adopted a naming standard that was voted and agreed upon. We're sticking with it.
